real_time/add-dynamic-entity-layer/src/main/java/com/esri/samples/add_dynamic_entity_layer/AddDynamicEntityLayerSample.java

```diff
@@ -0,0 +1,291 @@
+      mapView = new MapView();
+      mapView.setMap(map);
+
+      // create an envelope around Sandy, Utah to be the extent used by the dynamic entity filter and viewpoint
+      var utahSandyEnvelope = new Envelope(new Point(-112.110052, 40.718083, SpatialReferences.getWgs84()),
+        new Point(-111.814782, 40.535247, SpatialReferences.getWgs84()));
+
+      // set the viewpoint to the map view
+      mapView.setViewpoint(new Viewpoint(utahSandyEnvelope));
+
+      // create a graphics overlay and add a graphic to show the envelope's extent using a simple line symbol
+      var borderLineSymbol = new SimpleLineSymbol(SimpleLineSymbol.Style.DASH, Color.RED, 2);
+      var graphicsOverlay = new GraphicsOverlay();
+      var graphic = new Graphic(utahSandyEnvelope, borderLineSymbol);
+      graphicsOverlay.getGraphics().add(graphic);
+
+      // add the graphics overlay to the map view
+      mapView.getGraphicsOverlays().add(graphicsOverlay);
+
+      // create a controls vbox containing the UI components
+      setUpControlsVBox();
+
+      // create an ArcGIS stream service using a stream service URL
+      var streamServiceUrl = "https://realtimegis2016.esri.com:6443/arcgis/rest/services/SandyVehicles/StreamServer";
+      var streamService = new ArcGISStreamService(streamServiceUrl);
+
+      // add a filter for the data source to limit the amount of data received by the application
+      var streamServiceFilter = new ArcGISStreamServiceFilter();
+      streamServiceFilter.setGeometry(utahSandyEnvelope);
+      streamServiceFilter.setWhereClause("speed > 0");
+      streamService.setFilter(streamServiceFilter);
+
+      // set a duration of five minutes for how long observation data is stored in the data source
+      streamService.getPurgeOptions().setMaximumDuration(5);
+
+      // create a dynamic entity layer to display the data from the stream service
+      dynamicEntityLayer = new DynamicEntityLayer(streamService);
+
+      // create unique values for the different snow plow operating agencies represented in the data
+      // create a blue symbol to use as the default
+      var entityBlueSymbol = new SimpleMarkerSymbol(SimpleMarkerSymbol.Style.CIRCLE, Color.BLUE, 8);
+
+      // create a pink unique value for the agency represented by value "3"
+      var entityPinkSymbol = new SimpleMarkerSymbol(SimpleMarkerSymbol.Style.CIRCLE, Color.PINK, 8);
+      var entityPinkValue = new UniqueValueRenderer.UniqueValue("", "", entityPinkSymbol,
+        List.of(3), List.of(entityBlueSymbol));
+
+      // create a lime unique value for the agency represented by value "4"
+      var entityLimeSymbol = new SimpleMarkerSymbol(SimpleMarkerSymbol.Style.CIRCLE, Color.LIME, 8);
+      var entityLimeValue = new UniqueValueRenderer.UniqueValue("", "", entityLimeSymbol,
+        List.of(4), List.of(entityBlueSymbol));
+
+      // create a unique value renderer using unique values for the different snow plow agencies represented in the data
```

Comment:
```suggestion
      // create a unique value renderer using the configured unique values
```
Think I'd made this too wordy, sorry.